shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/throttler/prometheus/PrometheusThrottler.java

```diff
@@ -32,57 +31,76 @@
  * AutoInhibit{environment="e",job="s"}
  * </pre>
  *
- * where <i>e</i> matches the environment variable <i>SHESMU_ENVIONMENT</i> (or
- * is absent) and <i>s</i> is one of the services specified by the action.
+ * where <i>e</i> matches the environment specified in the configuration file
+ * (or is absent) and <i>s</i> is one of the services specified by the action.
  */
 @MetaInfServices(Throttler.class)
 public class PrometheusThrottler implements Throttler {
-	private static final CloseableHttpClient HTTP_CLIENT = HttpClients.createDefault();
-	private static final Gauge lastUpdated = Gauge
-			.build("shesmu_prom_throttler_last_update", "The last time the data was fetched.").register();
+	private class Endpoint extends AutoUpdatingJsonFile<Configuration> {
+
+		private Optional<Configuration> configuration = Optional.empty();
+
+		public Endpoint(Path fileName) {
+			super(fileName, Configuration.class);
+		}
+
+		public Pair<String, Map<String, String>> configuration() {
+			final Map<String, String> properties = new TreeMap<>();
+			configuration.ifPresent(c -> {
+				properties.put("address", c.getAlertmanager());
+				properties.put("environment", c.getEnvironment());
+			});
+
+			return new Pair<>("Prometheus Throttler: " + fileName().toString(), properties);
+		}
+
+		public boolean isOverloaded(Set<String> services) {
+			return configuration.flatMap(configuration -> //
+			CACHE.get(configuration.getAlertmanager())//
+					.map(l -> l.stream()//
+							.anyMatch(alert -> alert.matches(configuration.getEnvironment(), services))))
+					.orElse(false);
+		}
 
-	private static final Gauge ok = Gauge
-			.build("shesmu_prom_throttler_status", "Whether the last call to Alert Manager was successful.").register();
+		@Override
+		protected Optional<Integer> update(Configuration value) {
+			configuration = Optional.of(value);
+			return Optional.empty();
```

Comment:
Why does it return `Optional.empty()`?

Reply:
That's if you succeeded. Otherwise, the number of minutes to try again.